### PR TITLE
Added LiveFeedScreen page

### LINKED ISSUE #2091: Hub Camera Monitoring Screen
**Describe the bug**  Modified on 19/05/2023

Currently, on the platform, we do not have a page similar to the Central CNS page where we can view all the camera feed inside a facility at once.

**Steps to view the central CNS page**

1. go to https://care.coronasafe.in/ and log in with the credentials in the readme
2. Search for the name of the "Critical Care" facility and click on the CNS button.
3. The Page you are viewing is the central view with show vital monitors of all patient inside the facility

**Expected Solution**

Create a Screen that can View multiple Live Camera Feeds

- [ ] Work on the LiveFeedTile Component, Make it a simple Tile (Optional: Hovering on it the presets can show up.)
- [ ] Create a local storage that can store a set of asset_id for the specific hub device
- [ ] List the list of asset_ids as LiveFeedTiles as the Hub View
- [ ] More UX tweaks to Open the right Patient from the Hub (Wishlist)

### PR DESCRIPTION
**Pull request by kshitijv256**
### <samp>🤖 Generated by Copilot at 6b62f09</samp>

This pull request adds a new feature to the facility dashboard, which allows users to view the live camera feed of the facility and its patient asset beds. It creates a new `LiveFeedScreen` component, a new route, and some TypeScript fixes. It also imports the `ConsultationModel` type to the `src/Redux/api.tsx` file.

## Proposed Changes

- Fixes #2091 
- picking up from PR #6005 
- added a livefeed monitoring option in facility homepage
- livefeedtiles size and layout changes according to screen size
- on hovering any tile control options show up in

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers
@gigincg 

## Screenshots

imageedit_1_6895600569
Screenshot from 2023-10-31 13-43-33
Screenshot from 2023-10-31 13-45-11

### CODE REVIEW

**Comment by nihal467** [comment]
@kshitijv256 add screenshot / video of what you have made in the PR for QA testing

**Comment by kshitijv256** [comment]
> @kshitijv256 add screenshot / video of what you have made in the PR for QA testing

updated PR description with screenshots

**Comment by kshitijv256** [comment]
I Have not added user access control, should be better to add in a separate PR
> changed UI as livefeed in consultation

**Comment by nihal467** [comment]
@kshitijv256 
image

the tooltip is overlapping with the sidebar

File: src/Components/Facility/LiveFeedTile.tsx [hunk 1]
```diff
@@ -0,0 +1,1045 @@
...
+import useWindowDimensions from "../../Common/hooks/useWindowDimensions";
+import CareIcon from "../../CAREUI/icons/CareIcon";
+import ConfirmDialog from "../Common/ConfirmDialog";
+import { FieldLabel } from "../Form/FormFields/FormField";
+import useFullscreen from "../../Common/hooks/useFullscreen";
+import { FeedCameraPTZHelpButton } from "./Consultations/Feed";
+
+const LiveFeed = (props: any) => {
+  const middlewareHostname =
+    props.middlewareHostname || "dev_middleware.coronasafe.live";
```
Comment (rithviknishad) [inline src/Components/Facility/LiveFeedTile.tsx:449]: We seem to be duplicating this in a couple of places by now. This logic seems hard to maintain across multiple places.

**Review by rithviknishad — COMMENTED** [review]
Too much code duplication.

File: src/Components/Facility/LiveFeedTile.tsx [hunk 2]
```diff
@@ -0,0 +1,1045 @@
...
+// import ReactPlayer from "react-player";
+// import { getAsset, listAssetBeds } from "../../Redux/actions";
+// import { statusType, useAbortableEffect } from "../../Common/utils";
+// import { useTranslation } from "react-i18next";
+// import useFullscreen from "../../Common/hooks/useFullscreen.js";
+// interface LiveFeedTileProps {
+//   assetId: string;
+// }
+
+// interface CameraPosition {
```
Comment (rithviknishad) [inline src/Components/Facility/LiveFeedTile.tsx:14]: So much unused commented code. Why?

File: src/Components/Facility/LiveFeedScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+import { Popover, Transition } from "@headlessui/react";
+import { FieldLabel } from "../Form/FormFields/FormField";
+import CheckBoxFormField from "../Form/FormFields/CheckBoxFormField";
+import { useTranslation } from "react-i18next";
+import { SortOption } from "../Common/SortDropdown";
+import { SelectFormField } from "../Form/FormFields/SelectFormField";
+import LiveFeedTile from "./LiveFeedTile";
+import { getCameraConfig } from "../../Utils/transformUtils";
+import { getPermittedFacility, listAssets } from "../../Redux/actions";
+import { useDispatch } from "react-redux";
```
Comment (rithviknishad) [inline src/Components/Facility/LiveFeedScreen.tsx:23]: You are not supposed to be using `useDispatch` in newer codes.

**Review by rithviknishad — APPROVED** [review]
Momentarily approving. Will create a separate issue to clean this up and improve the UI.

**Comment by kshitijv256** [comment]
@nihal467 
> the tooltip is overlapping with the sidebar

 Screenshot from 2023-11-06 22-31-23

Moved controls to center of tile

File: package-lock.json
Comment (rithviknishad) [inline package-lock.json:1]: @kshitijv256 you'll have to discard your package-lock changes which was added in the recent commit.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by khavinshankar** [comment]
@rithviknishad can you re-review this once

**Review by rithviknishad — APPROVED** [review]

**Comment by rithviknishad** [comment]
Done @khavinshankar

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**